File: nodeJs/sources/testes2.js
fs=require('fs');
path=require('path');

imagens=120;
amostra=new Array(imagens).fill(0);

//contabilizando acertos e erros
circulocerto=0,circuloerrado=0,quadradocerto=0,quadradoerrado=0;

for(i=0;i<imagens;i++){
	buf=fs.readFileSync(`imagens/amostra${i.toString().padStart(4,'0')}.bmp`);
	//BMP header tem 54 bytes
	//imagem 60x60x3
	soma=0;
	for(j=54;j<buf.length;j++)soma+=buf[j];
	amostra[i]=soma;
}

//inicializa o bias com valor buscado do txt
bias=parseFloat(fs.readFileSync('bias2.txt','utf-8'));

for(x=0;x<1000;x++){ //testa centenas de vezes
	img=parseInt(Math.random()*imagens); //uma imagem qualquer

	if(amostra[img]>bias){
		if(img<imagens/2)circulocerto++;
		else circuloerrado++;
	}else{
		if(img>imagens/2)quadradocerto++;
		else quadradoerrado++;
	}
}

console.log(`\n${circulocerto + circuloerrado + quadradocerto + quadradoerrado} tentativas:`);
console.log(`Esperado circulo dado circulo: ${circulocerto}`);
console.log(`Esperado circulo dado quadrado: ${circuloerrado}`);
console.log(`Esperado quadrado dado quadrado: ${quadradocerto}`);
console.log(`Esperado quadrado dado circulo: ${quadradoerrado}\n`);

File: python/sources/testes2.py
import random
import os

imagens = 120
amostra = [0] * imagens

# Contabilizando acertos e erros
circulocerto = circuloerrado = quadradocerto = quadradoerrado = 0

for i in range(imagens):
    with open(f"imagens/amostra{i:04d}.bmp", "rb") as f:
        buf = f.read()
    amostra[i] = sum(buf[54:])

# Lê bias2.txt
with open('bias2.txt', 'r', encoding='utf-8') as f:
    bias = float(f.read().strip())

for x in range(1000):  # testa 1000 vezes
    img = random.randint(0, imagens - 1)  # uma imagem qualquer

    if amostra[img] > bias:
        if img < imagens // 2:
            circulocerto += 1
        else:
            circuloerrado += 1
    else:
        if img > imagens // 2:
            quadradocerto += 1
        else:
            quadradoerrado += 1

print(f"\n{circulocerto + circuloerrado + quadradocerto + quadradoerrado} tentativas:")
print(f"Esperado circulo dado circulo: {circulocerto}")
print(f"Esperado circulo dado quadrado: {circuloerrado}")
print(f"Esperado quadrado dado quadrado: {quadradocerto}")
print(f"Esperado quadrado dado circulo: {quadradoerrado}\n")
